Echo.Core/src/Textures/Colors/RGBA128.Parser.cs

```diff
@@ -141,9 +145,84 @@ bool ParseHex(out RGBA128 result)
 			static int CombineHex(int hex0, int hex1) => hex0 * 16 + hex1;
 		}
 
-		bool ParseRGB(out RGBA128 result) => throw new NotImplementedException();
+		bool ParseRGB(out RGBA128 result)
+		{
+			int index = 0;
+
+			int r = 0, g = 0, b = 0, a = 255;
+
+			if (findNextInt(content, ref index, ref r) &&
+				findNextInt(content, ref index, ref g) &&
+				findNextInt(content, ref index, ref b))
+			{
+				if (findNextInt(content, ref index, ref a)) return YieldError(out result);
+
+
+				if (IsChannelInRange(r) &&
+					IsChannelInRange(g) &&
+					IsChannelInRange(b) &&
+					IsChannelInRange(a))
+				{
+					result = new RGBA128(r / 255f, g / 255f, b / 255f, a / 255f);
+					return true;
+				}
+
+			}
+			return YieldError(out result);
+
+			static bool IsChannelInRange(int channel) => channel > 0 && channel <= 255;
+			static bool findNextInt(ReadOnlySpan<char> src, ref int index, ref int result)
+			{
+				int len = 0;
+
+				while (!char.IsDigit(src[index])) if (src.Length <= index++) return false; // find first digit
+				while (char.IsDigit(src[index + len]) && (index + len) < (src.Length - 1)) len++; // get the length of the number
+
+				if (len == 0) return false;
+
+				ReadOnlySpan<char> num = src.Slice(index, len);
+				result = int.Parse(num);
+				index += len;
+				return true;
+			}
+		}
+
+		bool ParseHDR(out RGBA128 result)
+		{
+			int index = 0;
+			float r = 0f, g = 0f, b = 0f, a = 1f;
+
+			if (findNextFloat(content, ref index, ref r) &&
+				findNextFloat(content, ref index, ref g) &&
+				findNextFloat(content, ref index, ref b))
+			{
+				if (findNextFloat(content, ref index, ref a)) return YieldError(out result);
 
-		bool ParseHDR(out RGBA128 result) => throw new NotImplementedException();
+				result = new RGBA128(r, g, b, a);
+				return true;
+			}
+			return YieldError(out result);
+
+			static bool findNextFloat(ReadOnlySpan<char> src, ref int index, ref float result)
+			{
+				int len = 0;
+
+				while (!char.IsDigit(src[index])) if (src.Length <= index++) return false; // find first digit
+				while ((char.IsDigit(src[index + len]) || src[index + len].Equals('.')) && (index + len) < (src.Length - 1)) len++; // get the length of the number
+
+				if (len == 0) return false;
+
+				ReadOnlySpan<char> num = src.Slice(index, len);
+				float tempRes = 0f;
+				bool parsed = float.TryParse(num, result: out tempRes);
+
+				if (parsed) result = tempRes;
+				else return false;
+				index += len;
+
+				return true;
+			}

```

Comment:
Please also revise this method based on the comment for the `findNextInt` method in `ParseRGB`.